android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.java

```diff
@@ -660,65 +660,70 @@ private Boolean installClient() {
-		// and assets directory select accordingly
-		String source = "";
-		if(executable) source = getAssestsDirForCpuArchitecture() + file;
-		else source = file;
-		
-		try {
-			if(Logging.ERROR) Log.d(Logging.TAG, "installing: " + source);
+        Boolean success = false;
+        byte[] b = new byte [1024];
+        int count;
+		
+        // If file is executable, cpu architecture has to be evaluated
+        // and assets directory select accordingly
+        String source = "";
+        if(executable) source = getAssestsDirForCpuArchitecture() + file;
+        else source = file;
+		
+        try {
+            if(Logging.ERROR) Log.d(Logging.TAG, "installing: " + source);
 			
-    		File target = new File(boincWorkingDir + file);
-    		
-    		// Check path and create it
-    		File installDir = new File(boincWorkingDir);
-    		if(!installDir.exists()) {
-    			installDir.mkdir();
-    			installDir.setWritable(true); 
-    		}
+            File target = new File(boincWorkingDir + file);
     		
-    		if(target.exists()) {
-    			if(override) target.delete();
-    			else {
-    				if(Logging.DEBUG) Log.d(Logging.TAG,"skipped file, exists and ovverride is false");
-    				return true;
-    			}
-    		}
+            // Check path and create it
+            File installDir = new File(boincWorkingDir);
+            if(!installDir.exists()) {
+                if (!installDir.mkdir()) {
+                    if(Logging.ERROR) Log.d(Logging.TAG,"Monitor.installFile(): mkdir() was not successful.");
+                }
+
+                if (!installDir.setWritable(true)) {
+                    if(Logging.ERROR) Log.d(Logging.TAG,"Monitor.installFile(): setWritable() was not successful.");
+                }
+            }
+
+            if(target.exists()) {
+                if(override) {
+                    if (!target.delete()) {
+                        if(Logging.ERROR) Log.d(Logging.TAG,"Monitor.installFile(): delete() was not successful.");
+                    }
+                } else {
+                    if(Logging.DEBUG) Log.d(Logging.TAG,"skipped file, exists and ovverride is false");
+                    return true;
+                }
+            }
     		
-    		// Copy file from the asset manager to clientPath
-    		InputStream asset = getApplicationContext().getAssets().open(source); 
-    		OutputStream targetData = new FileOutputStream(target); 
-    		while((count = asset.read(b)) != -1){ 
-    			targetData.write(b, 0, count);
-    		}
-    		asset.close(); 
-    		targetData.flush(); 
-    		targetData.close();
+            // Copy file from the asset manager to clientPath
+            InputStream asset = getApplicationContext().getAssets().open(source);
+            OutputStream targetData = new FileOutputStream(target);
+            while((count = asset.read(b)) != -1){
+                targetData.write(b, 0, count);
+            }
+            asset.close();
+            targetData.flush();
```

Comment:
I believe `flush()` function right before `close()` is redundant and could be safely removed

Reply:
Checked, Java's `close()` flush first, then close.